addons/csv-data-importer/import_plugin.gd

```diff
@@ -93,15 +94,25 @@ func _import(source_file, save_path, options, platform_variants, gen_files):
 	var lines = []
 	while not file.eof_reached():
 		var line = file.get_csv_line(delim)
-		if options.detect_numbers and (not options.headers or lines.size() > 0):
+		if not options.headers or lines.size() > 0:
 			var detected := []
 			for field in line:
-				if not options.force_float and field.is_valid_int():
-					detected.append(int(field))
-				elif field.is_valid_float():
-					detected.append(float(field))
-				else:
-					detected.append(field)
+				if options.detect_numbers and field.is_valid_float(): # Any valid integer is also a valid float
+					if not options.force_float and field.is_valid_int():
+						detected.append(int(field))
+						continue
+					elif field.is_valid_float():
```

Comment:
```suggestion
						continue
					if field.is_valid_float():
```

`elif` is not necessary since `continue` returns immediately.

The same applies to the `elif` below too.